Add DateTime datetime type entities to docs (#3784)
OttoWinter/esphomedocs: components/datetime/index.rst

  See https://developers.home-assistant.io/docs/core/entity/#generic-properties
  for a list of available options. Requires Home Assistant 2021.11 or newer.
  Set to ``""`` to remove the default entity category.
- **time_id** (**Required**, :ref:`config-id`): THe ID of the time entity. Automatically set

>   to the id of a time component if a single one is defined.

Reviewer: ```suggestion
- **time_id** (**Required**, :ref:`config-id`): The ID of the time entity. Automatically set
  to the ID of a time component if only a single one is defined.
```

DateTime Automation
-------------------

.. _datetime-datetime_set_action:

``datetime.datetime.set`` Action
********************************

This is an :ref:`Action <config-action>` for setting a datetime datetime state.
The ``datetime`` provided can be in one of 3 formats:

.. code-block:: yaml

    # String datetime
    - datetime.time.set:
        id: my_datetime
        datetime: "2024-12-31 12:34:56"

    # Individual datetime parts
    - datetime.datetime.set:
        id: my_datetime
        datetime:
          year: 2024
          month: 12
          day: 31
          hour: 12
          minute: 34
          second: 56

    # Using a lambda
    - datetime.datetime.set:
        id: my_datetime
        datetime: !lambda |-
          // Return an ESPTime struct
          return {.second: 56, .minute: 34, .hour: 12, .day_of_month: 31, .month: 12, .year: 2024};

Configuration variables:

- **id** (**Required**, :ref:`config-id`): The ID of the datetime to set.
- **datetime** (**Required**, string, datetime parts, :ref:`templatable <config-templatable>`):
  The value to set the datetime to.

.. _datetime-datetime-lambda_calls:

> lambda calls

Reviewer: ```suggestion
Lambda calls
```

************

From :ref:`lambdas <config-lambda>`, you can call several methods on all datetimes to do some

> advanced stuff (see the full API Reference for more info).

Reviewer: Haha, I just don't like the word "stuff" 😅 

```suggestion
For more complex use cases, several methods are available for use on datetimes from within :ref:`lambdas <config-lambda>`. See the full API Reference for more information.
```
